src/stories/components/user-card/user-card.tsx

```diff
@@ -1,44 +1,68 @@
 import styled from '@emotion/styled';
-import React from 'react';
+import React, { useState } from 'react';
 import lightTheme from '../../../../styles/theme/light';
+import { useAuthContext } from '../../../core/context/AuthContext';
 import { useThemeContext } from '../../../core/context/ThemeContext';
 import { ButtonType } from '../../../core/enums/button-type.enum';
-import { RoleEnum } from '../../../core/enums/role.enum';
+import { RoleUserDTO } from '../../../core/models/dto/role.dto';
 import { getColorRole } from '../../../core/utils/color.utils';
+import { getCorrectRoleApi } from '../../../core/utils/string.utils';
+import { ENABLE_DISABLE_USER_REQUEST } from '../../containers/auth/enable-disable-accounts/enable-disable.api';
 import ControlledSwitches from '../button/switch-toogle/switch-component';
 import { CustomButton } from '../custom-button/custom-button';
 import AvatarPlaceholder from '../svg/avatar-placeholder';
 
 interface Props {
-  role: RoleEnum;
+  role: RoleUserDTO;
   user: string;
   checked: boolean;
-  handleChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
-  handleDeleteAccount?: () => void;
+  id: string;
+  handleDeleteAccount?: (id: string) => void;
   handleViewProfile?: () => void;
+  handleGoProfileView?: (id: string) => void;
 }
 
 const UserCard = ({
   role,
   checked,
-  handleChange,
   user,
+  id,
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   handleDeleteAccount = () => {},
   // eslint-disable-next-line @typescript-eslint/no-empty-function
   handleViewProfile = () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  handleGoProfileView = () => {},
```

Comment:
the props allow undefined functions, this functions should be handled later if they are undefined or not instead of using default empty functions